dask/array/creation.py

```diff
@@ -143,3 +144,36 @@ def arange(*args, **kwargs):
         elem_count += bs
 
     return Array(dsk, name, chunks, dtype=dtype)
+
+
+def diag(v):
+    """Construct a diagonal array, with ``v`` on the diagonal.
+
+    TODO: implement diagonal extraction, and the ``k``th diagonal keyword arg.
```

Comment:
Request remove TODO from the docstring.  Explaining the restrictions sounds good.  The term TODO seems like it belongs in developer  and not user comments though.